app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -7,20 +7,47 @@ class Bitrise
       private
 
       def parse_workflow(wf_name, wf_config)
-        # Extend for other types of steps - approvals/waits for example.
-        BK::Compat::CommandStep.new(label: wf_name, key: wf_name).tap do |cmd_step|
-          wf_config['steps'].each do |step|
-            step_key, step_config = step.first
-            step_inputs = step_config['inputs'].reduce({}, :merge)
-            cmd_step << load_workflow_step(step_key, step_inputs)
-          end
+        agents = extract_agents(wf_config)
+        cmd_step = create_command_step(wf_name, agents)
+
+        wf_config['steps'].each do |step|
+          step_key, step_config = step.first
+          translated_step = translate_step(step_key, step_config)
+          add_step_to_command_step(cmd_step, translated_step)
         end
+
+        cmd_step
+      end
+
+      def extract_agents(wf_config)
+        wf_config.include?('stack') ? { stack: wf_config['stack'] } : {}
+      end
+
+      def create_command_step(wf_name, agents)
+        BK::Compat::CommandStep.new(label: wf_name, key: wf_name, agents: agents)
+      end
+
+      def translate_step(step_key, step_config)
+        step_inputs = extract_step_inputs(step_config)
+        translated_step = load_workflow_step(step_key, step_inputs)
+        set_agents_for_step(translated_step, step_config)
+        translated_step
```

Comment:
it is a lot more idiomatic to use `tap`:

```suggestion
        load_workflow_step(step_key, step_inputs).tap do |translated_step|
            translated_step.agents['stack'] = step_config['stack'] if step_config.include?('stack')
        end
```